### ISSUE TITLE
Python module doesn't appear to initialise google logging

### ISSUE THREAD

**Issue by ferrouswheel** [BVLC/caffe#4715]
People may want to control the logging of pycaffe using the google logging environment variables.

e.g.  you can set them in `os.environ` before calling import caffe:

```
os.environ['GLOG_minloglevel'] = "0"
os.environ['GLOG_log_dir'] = "."
os.environ['GLOG_logbufsecs'] = "1"
os.environ['GLOG_logtostderr']="0"
os.environ['GLOG_stderrthreshold'] = "3"

import caffe
```

Weirdly, the only one that would be respected was `minloglevel`, and I tracked this down to pycaffe not explicitly initialising the google logging (as far as I could tell).

I found that if I added `google::InitGoogleLogging("pycaffe");` to https://github.com/BVLC/caffe/blob/master/python/caffe/_caffe.cpp#L282 then pycaffe would start respecting these logging environment variables.

Without this, I don't get any log file written, to a temp directory or otherwise.

**Comment by xdshang** [comment]
Great! It works for me.

**Comment by liubarca** [comment]
It is very useful.

**Comment by shelhamer** [comment]
Initializing logging from Python was included in #4563 by https://github.com/BVLC/caffe/pull/4563/commits/2317fa19d3f5a65cb22adcbd3792ea248996744e

Thanks for pointing this out @ferrouswheel
